### ISSUE TITLE
Proposal: warn against unnecessary use of module.exports

### ISSUE THREAD

**Issue by ronkorving** [eslint/eslint#3637]
Sometimes I see Node/CommonJS developers write:

```js
module.exports.foo = function (a, b) {
};
```

for no apparent reason. The only time you should be referring to module.exports imho is when you're overwriting it with a new value. In all other cases, you should just be assigning properties on `exports`. I don't know where this weird pattern originated to be honest, but since I see multiple people make this mistake, perhaps a rule against it would make sense?

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by nzakas** [comment]
Thanks for the suggestion. I'm not sure it's worth having a core rule for this, but I'd encourage you to explore creating a [custom rule](http://eslint.org/docs/developer-guide/working-with-rules) if you'd like such a rule.

**Comment by xjamundx** [comment]
@ronkorving I'd be happy to add this rule to https://github.com/xjamundx/eslint-plugin-modules

**Comment by ronkorving** [comment]
@xjamundx thanks, done :)
